src/mdns/mdns_mdnssd.cpp

```diff
@@ -629,6 +670,119 @@ void PublisherMDnsSd::DnssdHostRegistration::HandleRegisterResult(DNSRecordRef a
     }
 }
 
+otbrError PublisherMDnsSd::DnssdKeyRegistration::Register(void)
+{
+    DNSServiceErrorType       dnsError = kDNSServiceErr_NoError;
+    DnssdServiceRegistration *serviceReg;
+
+    otbrLogInfo("Registering new key %s", mName.c_str());
+
+    serviceReg = static_cast<DnssdServiceRegistration *>(GetPublisher().FindServiceRegistration(mName));
+
+    if ((serviceReg != nullptr) && (serviceReg->mServiceRef != nullptr))
+    {
+        otbrLogInfo("Key %s is being registered as a record of an existing service registration", mName.c_str());
+
+        dnsError = DNSServiceAddRecord(serviceReg->mServiceRef, &mRecordRef, kDNSServiceFlagsUnique,
+                                       kDNSServiceType_KEY, mKeyData.size(), mKeyData.data(), /* ttl */ 0);
+
+        VerifyOrExit(dnsError == kDNSServiceErr_NoError);
+
+        mRelatedServiceReg         = serviceReg;
+        serviceReg->mRelatedKeyReg = this;
+
+        if (mRelatedServiceReg->IsCompleted())
+        {
+            HandleRegisterResult(kDNSServiceErr_NoError);
+        }
+
+        // If related service registration is not yet finished,
+        // we wait for service registration completion to signal
+        // key record registration as well.
+    }
+    else
+    {
+        otbrLogInfo("Key %s is being registered individually", mName.c_str());
+
+        dnsError = GetPublisher().CreateSharedHostsRef();
+        VerifyOrExit(dnsError == kDNSServiceErr_NoError);
+
+        dnsError = DNSServiceRegisterRecord(GetPublisher().mHostsRef, &mRecordRef, kDNSServiceFlagsUnique,
+                                            kDNSServiceInterfaceIndexAny, MakeFullKeyName(mName).c_str(),
+                                            kDNSServiceType_KEY, kDNSServiceClass_IN, mKeyData.size(), mKeyData.data(),
+                                            /* ttl */ 0, HandleRegisterResult, this);
```

Comment:
Why using 0 as the TTL?

Reply:
Zero TTL passed to `DNSServiceRegisterRecord()` API instructs it to use its own recommended default value. Here is the documenation:
```
 *
 *  @param ttl
 *                  The time to live of the resource record, in seconds.
 *                  Most clients should pass 0 to indicate that the system should
 *                  select a sensible default value.
 *
```

We use the same approach (i.e., we pass zero as TTL) when registering AAAA record in `HostRegisteration::Register()` to implement `PublishHost()`.